internal/controllers/subnet/status.go

```diff
@@ -50,21 +50,40 @@ func (subnetStatusWriter) ResourceAvailableStatus(orcObject orcObjectPT, osResou
 
 func (subnetStatusWriter) ApplyResourceStatus(log logr.Logger, osResource *osResourceT, statusApply statusApplyPT) {
 	status := orcapplyconfigv1alpha1.SubnetResourceStatus().
-		WithName(osResource.Name).
-		WithDescription(osResource.Description).
-		WithIPVersion(int32(osResource.IPVersion)).
-		WithCIDR(osResource.CIDR).
-		WithGatewayIP(osResource.GatewayIP).
 		WithDNSPublishFixedIP(osResource.DNSPublishFixedIP).
 		WithEnableDHCP(osResource.EnableDHCP).
-		WithNetworkID(osResource.NetworkID).
-		WithProjectID(osResource.ProjectID).
 		WithRevisionNumber(int64(osResource.RevisionNumber)).
-		WithIPv6AddressMode(osResource.IPv6AddressMode).
-		WithIPv6RAMode(osResource.IPv6RAMode).
 		WithTags(osResource.Tags...).
 		WithDNSNameservers(osResource.DNSNameservers...)
 
+	if osResource.Name != "" {
+		status.WithName(osResource.Name)
+	}
+	if osResource.Description != "" {
+		status.WithDescription(osResource.Description)
+	}
+	if osResource.IPVersion > 0 {
+		status.WithIPVersion(int32(osResource.IPVersion))
+	}
+	if osResource.CIDR != "" {
+		status.WithCIDR(osResource.CIDR)
+	}
+	if osResource.GatewayIP != "" {
+		status.WithGatewayIP(osResource.GatewayIP)
+	}
+	if osResource.NetworkID != "" {
+		status.WithNetworkID(osResource.NetworkID)
+	}
+	if osResource.ProjectID != "" {
+		status.WithProjectID(osResource.ProjectID)
+	}
+	if osResource.IPv6AddressMode != "" {
+		status.WithIPv6AddressMode(osResource.IPv6AddressMode)
+	}
+	if osResource.IPv6RAMode != "" {
+		status.WithIPv6RAMode(osResource.IPv6RAMode)
+	}
```

Comment:
I would personally apply some discretion here about what should reasonably be omitted if it's not set. My feeling is that the following fields should always be set even if they're empty, because empty values would be interesting:
* Name
* NetworkID
* ProjectID

and possibly Description?